tensorflow/c/experimental/filesystem/plugins/gcs/gcs_filesystem.cc

```diff
@@ -66,8 +67,42 @@ void ParseGCSPath(const std::string& fname, bool object_empty_ok,
 // SECTION 1. Implementation for `TF_RandomAccessFile`
 // ----------------------------------------------------------------------------
 namespace tf_random_access_file {
+typedef struct GCSFile {
+  const std::string bucket;
+  const std::string object;
+  gcs::Client* gcs_client;  // not owned
+} GCSFile;
 
-// TODO(vnvo2409): Implement later
+void Cleanup(TF_RandomAccessFile* file) {
+  auto gcs_file = static_cast<GCSFile*>(file->plugin_file);
+  delete gcs_file;
+}
+
+// TODO(vnvo2409): Adding cache.
+// `google-cloud-cpp` is working on a feature that we may want to use.
+// See https://github.com/googleapis/google-cloud-cpp/issues/4013.
+int64_t Read(const TF_RandomAccessFile* file, uint64_t offset, size_t n,
+             char* buffer, TF_Status* status) {
+  auto gcs_file = static_cast<GCSFile*>(file->plugin_file);
+  auto stream = gcs_file->gcs_client->ReadObject(
+      gcs_file->bucket, gcs_file->object, gcs::ReadRange(offset, offset + n));
+  TF_SetStatusFromGCSStatus(stream.status(), status);
+  if ((TF_GetCode(status) != TF_OK) &&
+      (TF_GetCode(status) != TF_OUT_OF_RANGE)) {
+    return -1;
+  }
+  int64_t read;
+  if (!absl::SimpleAtoi(stream.headers().find("content-length")->second,
```

Comment:
I dont know if we should use `absl::SimpleAtoi` or `std::stoll`. `std::stoll` throw an exception when no conversion could be performed. `absl::SimpleAtoi` is more convenient but I really do not want to use it because we have to add a dependecy `absl::strings`

Reply:
Google coding style enforces avoidance of exceptions.

It's ok to depend on `absl::strings`.